### PR TITLE
Harmony 1622 - Better job progress calculation

### PR DESCRIPTION
**Pull request by indiejames**
## Jira Issue ID
HARMONY-1622

## Description
This PR changes the way we compute job progress. In particular, it will provide incremental progress updates for jobs that use concise or some other aggregating service.

## Local Test Steps
Run this branch locally with l2ss and concise services deployed. Then execute the following query
```
 curl -Ln -bj "http://localhost:3000/C1234208438-POCLOUD/ogc-api-coverages/1.0.0/collections/bathymetry/coverage/rangeset?concatenate=true&subset=lon(-160%3A160)&subset=lat(-80%3A80)&skipPreview=true&maxResults=5&subset=time(%221981-01-02T00%3A00%3A00.000Z%22%3A%222020-01-02T01%3A00%3A00.000Z%22)"
```

Open the job in the workflow-ui and also open the job status page in a separate tab. As you see the work-items complete and refresh the status page you should see the job progress increase. It will jump to 33 after the queyr-cmr step, then bump up as each l2ss work-item completes until it reach 66. Then it should jump to 100 after the concise work-item completes.

## PR Acceptance Checklist
* [x] Acceptance criteria met
* [x] Tests added/updated (if needed) and passing
* [ ] ~Documentation updated (if needed)~

### CODE REVIEW

**Review by ygliuvt — APPROVED** [review]
Run the test locally and I see the progress change from 33% to 36%, 40%, 66%, etc. to 100%.

**Review by chris-durbin — COMMENTED** [review]
Looks good - tested it out locally. I think we want one change to significantly reduce the weighting of completing query-cmr steps. I was seeing some large requests (25+ minutes) going to 50% within a few seconds because the single query-cmr work item for the request completed.

File: services/harmony/app/backends/workflow-orchestration/work-item-updates.ts [hunk 1]
```diff
@@ -217,26 +215,19 @@ async function handleFailedWorkItems(
...
-    if (QUERY_CMR_SERVICE_REGEX.test(step.serviceID)) {
-      step.workItemCount = Math.ceil(job.numInputGranules / env.cmrMaxPageSize);
-    } else if (!step.hasAggregatedOutput) {
-      step.workItemCount = job.numInputGranules;
-    } else {
-      step.workItemCount = 1;
-    }
-    await step.save(transaction);
-  }
+  const step = await getWorkflowStepByJobIdStepIndex(transaction, job.jobID, 1);
```
[the remark this answers is not in the document]
Comment (indiejames) [reply services/harmony/app/backends/workflow-orchestration/work-item-updates.ts:228]: Done

File: services/harmony/app/backends/workflow-orchestration/work-item-updates.ts [hunk 2]
```diff
@@ -513,12 +506,16 @@ async function createNextWorkItems(
       return newItem;
     });
 
+    nextWorkflowStep.workItemCount += newItems.length;
     await incrementReadyCount(tx, workItem.jobID, nextWorkflowStep.serviceID, newItems.length);
     for (const batch of _.chunk(newItems, batchSize)) {
       await WorkItem.insertBatch(tx, batch);
-      logger.info('Queued new batch of work items.');
+      logger.info('Created new set of work items.');
```
[the remark this answers is not in the document]
Comment (indiejames) [reply services/harmony/app/backends/workflow-orchestration/work-item-updates.ts:513]: Done.

**Comment by indiejames** [comment]
I changed the weight on query-cmr to 0.1, which gives a more realistic estimate of the actual progress.

**Review by chris-durbin — APPROVED** [review]

**Pull request merged**